bundles/org.openhab.binding.gree/src/main/java/org/openhab/binding/gree/internal/discovery/GreeDeviceFinder.java

```diff
@@ -0,0 +1,179 @@
+                    InetAddress remoteAddress = receivePacket.getAddress();
+                    int remotePort = receivePacket.getPort();
+
+                    // Read the response
+                    // String modifiedSentence = new String(receivePacket.getData(), StandardCharsets.UTF_8);
+                    // GreeScanResponseDTO scanResponseGson = gson.fromJson(modifiedSentence,
+                    // GreeScanResponseDTO.class);
+                    String modifiedSentence = new String(receivePacket.getData(), StandardCharsets.UTF_8);
+                    StringReader stringReader = new StringReader(modifiedSentence);
+                    GreeScanResponseDTO scanResponseGson = gson.fromJson(new JsonReader(stringReader),
+                            GreeScanResponseDTO.class);
+
+                    // If there was no pack, ignore the response
+                    if (scanResponseGson.pack == null) {
+                        logger.debug("Invalid packet format, ignore");
+                        continue;
+                    }
+
+                    // Decrypt message - a a GreeException is thrown when something went wrong
+                    /*
+                     * scanResponseGson.decryptedPack = GreeCryptoUtil
+                     * .decryptPack(GreeCryptoUtil.getAESGeneralKeyByteArray(), scanResponseGson.pack);
+                     * String decryptedMsg = GreeCryptoUtil.decryptPack(GreeCryptoUtil.getAESGeneralKeyByteArray(),
+                     * scanResponseGson.pack);
+                     */
+                    String decryptedMsg = scanResponseGson.decryptedPack = GreeCryptoUtil
+                            .decryptPack(GreeCryptoUtil.getAESGeneralKeyByteArray(), scanResponseGson.pack);
+                    logger.debug("Response received from address {}: {}", remoteAddress.getHostAddress(), decryptedMsg);
+
+                    // Create the JSON to hold the response values
+                    scanResponseGson.packJson = gson.fromJson(decryptedMsg, GreeScanReponsePackDTO.class);
+
+                    // Now make sure the device is reported as a Gree device
+                    if (scanResponseGson.packJson.brand.equalsIgnoreCase("gree")) {
+                        // Create a new GreeDevice
+                        logger.debug("Discovered device at {}:{}", remoteAddress.getHostAddress(), remotePort);
+                        GreeAirDevice newDevice = new GreeAirDevice();
+                        newDevice.setAddress(remoteAddress);
+                        newDevice.setPort(remotePort);
+                        newDevice.setScanResponseGson(scanResponseGson);
+                        addDevice(newDevice);
+                    } else {
+                        logger.debug("Unit discovered, but brand is not GREE");
+                    }
+
+                    if (!scanNetwork) {
+                        break;
+                    }
+                } catch (SocketTimeoutException e) {
+                    break;
```

Comment:
Returning from a method explicitly should be preferred over returning implicitly.
```suggestion
                        return;
                    }
                } catch (SocketTimeoutException e) {
                    return;
```

Reply:
done